File: src/Human.java
import java.util.Random;

public class Human {
    private String fullName;   // ФИО
    private int age;           // Возраст
    private double speed;      // Текущая скорость
    private double x;          // координата X
    private double y;          // координата Y

    private Random rnd = new Random();

    public Human(String fullName, int age, double speed, double x, double y) {
        this.fullName = fullName;
        this.age = age;
        this.speed = speed;
        this.x = x;
        this.y = y;
    }

    public String getFullName() { return fullName; }
    public void setFullName(String fullName) { this.fullName = fullName; }

    public int getAge() { return age; }
    public void setAge(int age) { this.age = age; }

    public double getSpeed() { return speed; }
    public void setSpeed(double speed) { this.speed = speed; }

    public double getX() { return x; }
    public double getY() { return y; }

    public void move() {
        double angle = rnd.nextDouble() * 2 * Math.PI;
        double dx = speed * Math.cos(angle);
        double dy = speed * Math.sin(angle);
        this.x += dx;
        this.y += dy;
    }

    @Override
    public String toString() {
        return String.format("%s (age=%d) pos=(%.2f, %.2f) v=%.2f",
                fullName, age, x, y, speed);
    }
}

File: app/src/main/java/Movable/Movable/Human.kt
import kotlin.math.*
import kotlin.random.Random

open class Human(
    val fullName: String,
    val age: Int,
    override var speed: Double,
    override var x: Double = 0.0,
    override var y: Double = 0.0
) : Movable {

    override fun move(dt: Double): Pair<Double, Double> {
        if (dt <= 0.0) return x to y
        val theta = Random.nextDouble(0.0, 2 * Math.PI)
        val distance = speed * dt
        val dx = distance * cos(theta)
        val dy = distance * sin(theta)
        x += dx
        y += dy
        return x to y
    }

    override fun toString(): String {
        return "${this::class.simpleName}(name='$fullName', age=$age, speed=${"%.2f".format(speed)}, x=${"%.3f".format(x)}, y=${"%.3f".format(y)})"
    }
}
